### PR TITLE
Update `getNotifications` resolver for consistency

### PR DESCRIPTION
**Pull request by derweili**
## Summary

Addresses issue:

- #6163

## Relevant technical choices

- Followed the IB. Used `createSettingsStore` as reference.
- Updated failing unit tests for `createNotificationsStore`.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by jimmymadon — APPROVED** [review]
Thanks @derweili. LGTM!

File: assets/js/googlesitekit/data/create-notifications-store.test.js [hunk 1]
```diff
@@ -46,12 +42,16 @@ describe( 'createNotificationsStore store', () => {
 	} );
 
 	beforeEach( () => {
-		registry = createRegistry();
+		registry = createTestRegistry();
 
 		storeDefinition = createNotificationsStore( ...STORE_ARGS );
-		registry.registerStore( storeDefinition.STORE_NAME, storeDefinition );
+
+		store = registry.registerStore(
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/data/create-notifications-store.test.js:49]: Nice 👍

File: assets/js/googlesitekit/data/create-notifications-store.test.js [hunk 2]
```diff
@@ -46,12 +42,16 @@ describe( 'createNotificationsStore store', () => {
 	} );
 
 	beforeEach( () => {
-		registry = createRegistry();
+		registry = createTestRegistry();
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/data/create-notifications-store.test.js:45]: This isn't be necessary but also, as a store composition utility we want to test in a bit more isolation than the average partial. Please keep this as before.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @derweili – this is really close, just one thing to address.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @derweili !

**Pull request merged**